python/python/bystro/ancestry/define_callset.py

```diff
@@ -0,0 +1,141 @@
+
+logger = logging.getLogger(__name__)
+converter = get_lifter("hg19", "hg38")
+
+ILLUMINA_FILEPATH = DATA_DIR / "Human660W-Quad_v1_H.csv"
+AFFYMETRIX_FILEPATH = DATA_DIR / "Axiom_PMRA.na35.annot.csv"
+
+
+def get_watson_crick_complement(base: str) -> str:
+    """Calculate Watson-Crick Complement."""
+    wc_dict = {"A": "T", "T": "A", "G": "C", "C": "G"}
+    return wc_dict[base]
+
+
+def liftover_38_from_37(variant: str) -> str | None:
+    """Liftover a variant to genome build 38 from 37."""
+    chrom, pos, ref, alt = variant.split(":")
+    if "MT" in chrom:
+        return None
+    # liftover doesn't deal gracefully with MT variants, but we don't need them anyway
+    locations = converter[chrom][int(pos)]
+    if locations is None or len(locations) != 1:
+        logger.debug("Variant %s had a non-unique location, couldn't lift over", variant)
+        return None
+    chrom38, pos38, _strand38 = locations[0]
+    variant38 = ":".join([chrom38, str(pos38), ref, alt])
+    assert_true("lifted-over variant starts with 'chr'", variant38.startswith("chr"))
+    return variant38
+
+
+def _load_illumina_df() -> pd.DataFrame:
+    comment_rows = 7
+    columns_to_keep = ["Chr", "MapInfo", "SNP", "RefStrand"]
+    illumina_df = pd.read_csv(ILLUMINA_FILEPATH, skiprows=comment_rows)
+    assert_equals(
+        "Set of genome builds",
+        {37.1},
+        "actual set of genome builds",
+        set(illumina_df.GenomeBuild.dropna()),
+    )
+    return illumina_df[columns_to_keep].dropna()
+
+
+def load_illumina_variants() -> pd.Series:
+    """Load list of variants for illumina chip."""
+    illumina_df = _load_illumina_df()
+    illumina_variants = _get_variants_from_illumina_df(illumina_df)
+    assert_equals(
+        "number of illumina variants", 578822, "recovered number of variants", len(illumina_variants)
+    )
+    return illumina_variants
+
+
+def _get_variants_from_illumina_df(illumina_df: pd.DataFrame) -> pd.Series:
+    """Extract affymetrix variants and lift over."""
+    variants38 = []
+    for _i, row in tqdm.tqdm(illumina_df.iterrows(), total=len(illumina_df)):
+        chromosome = str(row.Chr)
+        position = str(int(row.MapInfo))
+        if match := re.match(r"\[([ACGT])/([ACGT])\]", str(row.SNP)):
+            allele1, allele2 = match.groups()
+        else:
+            continue
+        if row.RefStrand == "-":
+            allele1 = get_watson_crick_complement(allele1)
+            allele2 = get_watson_crick_complement(allele2)
+        variant37 = ":".join(["chr" + chromosome, position, allele1, allele2])
+        if not is_autosomal_variant(variant37):
+            continue
+        variants38.append(liftover_38_from_37(variant37))
+    illumina_variants = pd.Series(variants38)
+    liftover_failure_rate = illumina_variants.isna().mean()
+    logger.info("liftover failure rate: %1.2f%%", liftover_failure_rate * 100)
+    return illumina_variants.dropna()
+
+
+def _load_affymetrix_df() -> pd.DataFrame:
```

Comment:
This may be something for QC we develop later that integrates the annotator, but did we want to include something for the affymetrix manifest that includes checking whether or not the rsID and position matches since we identified incorrect positions for some of them?

Reply:
Agreed in general, think it's out of scope for this PR (and we have accuracy checks on the model in place so we know that the discordance between rsids and coordinates can't be *that* bad).  Added a todo.